packages/nimble-components/src/anchor-menu-item/index.ts

```diff
@@ -0,0 +1,110 @@
+export class AnchorMenuItem extends AnchorBase {
+    @attr({ mode: 'boolean' })
+    public disabled = false;
+
+    @observable
+    public anchor!: HTMLAnchorElement;
+
+    @observable
+    public startColumnCount: AnchorMenuItemColumnCount = 0;
+
+    // The following two handlers are workarounds for issues with anchor menu items in submenus.
+    // Events can bubble up the DOM and get handled by the menu item in the parent menu. When that happens,
+    // the menu item's handlers (FAST) return false and prevent the default action, i.e. navigation.
+
+    public clickHandler = (e: MouseEvent): boolean => {
+        e.stopImmediatePropagation();
+        return true;
+    };
+
+    public keydownHandler = (e: KeyboardEvent): boolean => {
+        if (e.defaultPrevented) {
+            return false;
+        }
+        switch (e.key) {
+            case keyEnter:
+                e.stopImmediatePropagation();
+                break;
+            default:
+        }
+        return true;
+    };
+
+    // The FAST Menu manages the `tabindex` of its menu items. Because we're setting `delegatesFocus=true`
+    // the focusable element is the anchor, so we forward `tabindex` to that element instead of the host.
+    // This prevents a problem where the focused anchor loses focus and the menu closes.
+    public override setAttribute(qualifiedName: string, value: string): void {
+        if (qualifiedName === 'tabindex') {
+            this.anchor.setAttribute(qualifiedName, value);
+        } else {
+            super.setAttribute(qualifiedName, value);
+        }
+    }
+
+    public override get tabIndex(): number {
+        return this.anchor.tabIndex;
+    }
+
+    public override set tabIndex(value: number) {
+        this.anchor.tabIndex = value;
+    }
+}
+
+const nimbleAnchorMenuItem = AnchorMenuItem.compose<AnchorOptions>({
+    baseName: 'anchor-menu-item',
+    template,
+    styles,
+    shadowOptions: {
+        delegatesFocus: true
+    }
+});
+
+DesignSystem.getOrCreate()
+    .withPrefix('nimble')
+    .register(nimbleAnchorMenuItem());
+export const anchorMenuItemTag = DesignSystem.tagFor(AnchorMenuItem);
+
+// This is a workaround for the fact that FAST's menu uses `instanceof MenuItem`
+// in their logic for indenting menu items. Since our AnchorMenuItem derives
+// from AnchorBase and not FAST's MenuItem, we need to change their MenuItem's
+// definition of `hasInstance` so that it includes our AnchorMenuItem, too.
+// The logic really only cares that the object has a `startColumnCount` member,
+// so we test for that.
+//
+// If/when we change FAST to test for the presence of `startColumnCount` instead
+// of using `instanceof MenuItem`, we can remove this workaround. Here is the
+// PR into FAST: https://github.com/microsoft/fast/pull/6667
+Object.defineProperty(FoundationMenuItem, Symbol.hasInstance, {
+    value(instance: unknown) {
+        return instance instanceof Object && 'startColumnCount' in instance;
+    }
+});
```

Comment:
With the imports:

```html
import { MenuItem as FoundationMenuItem } from '@microsoft/fast-foundation';
import { MenuItem as NimbleMenuItem } from '../menu-item';
```

We can have an implementation like:

```suggestion
Object.defineProperty(NimbleMenuItem, Symbol.hasInstance, {
    value(instance: unknown) {
        return instance instanceof FoundationMenuItem || instance instanceof AnchorMenuItem;
    }
});
```

The approach also avoids excessive deviation from the original behavior. It's important when overriding such low-level behavior to do it in a targeted way.

For example, the current implementation would have failed instance of checks for everything in base classes, i.e. `instanceof FoundationElement`, `instanceof HTMLElement`, etc. That's a very big deviation from the behavior of `instanceof`.

Reply:
Using the second implementation you linked. Works as expected.

Reply:
Sorry, re-opening again, realized an even simpler approach. Just re-use the original instanceof as part of the check.
It's the latest commit in the same branch

https://github.com/ni/nimble/compare/users/makinc/anchor-menu-item...mraj-anchor-menu-item-instanceof